App/views.py

```diff
@@ -0,0 +1,36 @@
+from django.http import HttpResponse
+from django.shortcuts import render
+import math
+import json
+import requests
+import toolforge
+
+def main2(request):
+    if(request.method=='POST'):
+        username = request.POST['username']
+        conn = toolforge.connect('kawiki_p')
+        with conn.cursor() as cursor:
+            s1 = cursor.execute("SELECT user_editcount FROM user WHERE user_name = %s",[username])
+            useredits_tup = cursor.fetchall()
+            s2 = cursor.execute("SELECT ss_total_edits FROM site_stats")
+            totaledits_tup =cursor.fetchall()
+            s3 = cursor.execute("SELECT 1+ (SELECT count(*) from user a WHERE a.user_editcount > b.user_editcount) as RNK FROM user b WHERE user_name = %s",[username])
+            userrank_tup = cursor.fetchall()
+            s4 = cursor.execute("select count(*) from user")
+            totalusers_tup = cursor.fetchall()
+        conn.close()
+
+        try:    
+            useredits = useredits_tup[0][0]
+            totaledits = totaledits_tup[0][0]
+            userrank = userrank_tup[0][0]
+            totalusers = totalusers_tup[0][0]
+            percentile = 100-((userrank*100)/(totalusers+1))
```

Comment:
I don't understand the logic here. In practice this will be the same as `100 - rating` (since `totalusers` is large so adding one doesn't matter). In theory, the `+1` seems wrong. The percentile is the smallest number N for which the user is included in the top N% of users - you order the users by editcount, split the list into 100 equal chunks, if the user is in the chunk with the largest editcount, they are in the first percentile, the next chunk is the second etc. So the top 1% is the users with rank 1..(totalusers/100), top 2% is (totalusers/100)..(2*totalusers/100) etc. So percentile is same as what you call `rating` here (or 100 minus that if you prefer - either could be called a percentile).

Reply:
I was calculating percentile on the basis no of editcounts , so the user with highest editcounts gets 99 percentile (since 99% of users lie below him) and not the basis of userrank. In which case, the percentile of the user with highest no of editcounts would we 1, so he would in top 1 percentile. 
I'll have updated it according to userrank.